wgsl/index.bs

```diff
@@ -3805,8 +3823,98 @@ shorten vector declarations based on the type or number of elements provided.
 If you want `vec4<f32>` you must provide 4 float values in the constructor.
 
 ## Precedence ## {#precedence}
-
-Issue: (dsinclair) Write out precedence rules. Matches c and glsl rules ....
+<table class='data precedence'>
+  <thead>
+    <tr>
+      <th>Precedence
+      <th>Operators
+      <th>Description
+      <th>Associativity
+  </thead>
+  <tr>
+    <th rowspan='1'>1 (highest)
+    <td> `(` `)`
+    <td> Grouping
+    <td rowspan='1'> Left-to-right
+  <tr>
+    <th colspan='1' rowspan='3'>2
+    <td> `!`
+    <td> Logical not
+    <td colspan='1' rowspan='3'>
+      Right-to-left
+  <tr>
+    <td> `~`
+    <td> Bitwise not
+  <tr>
+    <td> `-`
+    <td> Unary negation
+  <tr>
+    <th colspan='1' rowspan='3'>3
+    <td> `*`
+    <td> Multiplication
+    <td colspan='1' rowspan='18'>
+      Left-to-right
+  <tr>
+    <td> `/`
+    <td> Division
+  <tr>
+    <td> `%`
+    <td> Remainder
+  <tr>
+    <th colspan='1' rowspan='2'>4
+    <td> `+`
+    <td> Addition
+  <tr>
+    <td> `-`
+    <td> Subtraction
+  <tr>
+    <th colspan='1' rowspan='2'>5
+    <td> `<<`
+    <td> Bitwise left shift
+  <tr>
+    <td> `>>`
+    <td> Bitwise right shift
+  <tr>
+    <th colspan='1' rowspan='4'>6
+    <td> `<`
+    <td> Less than
+  <tr>
+    <td> `<=`
+    <td> Less than or equal
+  <tr>
+    <td> `>`
+    <td> Greater than
+  <tr>
+    <td> `>=`
+    <td> Greater than or equal
+  <tr>
+    <th colspan='1' rowspan='2'>7
+    <td> `==`
+    <td> Equality
+  <tr>
+    <td> `!=`
+    <td> Inequality
```

Comment:
Just occurred to me: I think we declared that `^` is for bitwise operations only, and for boolean xor you need `!=`?

In that case we should call it out here
```suggestion
    <td> Inequality, boolean xor
```

What do others think?

Reply:
as long as it's consistent with `&` and `|`... unless that consistency leads us to `^^` overload for `!=` :laughing: